shared/common-adapters/tab-bar.native.js

```diff
@@ -29,6 +29,18 @@ class SimpleTabBarButton extends Component<void, ItemProps, void> {
   }
 }
 
+const BadgeComponent = ({badgeNumber, badgePosition}: {badgeNumber: number, badgePosition: TabBadgePosition}) => {
```

Comment:
no i mean, we already pass in style props, just make a badgeStyle and pass that through.
this and the nav thing, these 'functions' are special functions for rendering react components. Should be capitalized and not called like {func(props} but <Func prop1={true} />. see https://facebook.github.io/react/docs/components-and-props.html#functional-and-class-components